plugin-mainmenu/actionview.cpp

```diff
@@ -70,16 +72,26 @@ namespace
 
         virtual QSize sizeHint(const QStyleOptionViewItem& option, const QModelIndex& index) const override
         {
-            //the XdgCachedMenuAction does load the icon upon showing its menu
-#ifdef HAVE_MENU_CACHE
             QIcon icon = index.data(Qt::DecorationRole).value<QIcon>();
+            //the XdgCachedMenuAction/XdgAction does load the icon upon showing its menu
+#ifdef HAVE_MENU_CACHE
             if (icon.isNull())
             {
                 XdgCachedMenuAction * cached_action = qobject_cast<XdgCachedMenuAction *>(qvariant_cast<QAction *>(index.data(ActionView::ActionRole)));
                 Q_ASSERT(nullptr != cached_action);
                 cached_action->updateIcon();
                 const_cast<QAbstractItemModel *>(index.model())->setData(index, cached_action->icon(), Qt::DecorationRole);
             }
+#else
```

Comment:
Did you test this w/o invoking the `updateIcon()`? As the `XdgAction` should be setting the icon in the creation time https://github.com/lxqt/libqtxdg/blob/12328819436faf21cc976d02ad614c92aea0ad40/qtxdg/xdgaction.cpp#L98 ... 
... or the problem is the "delayed" execution of `updateIcon()` and the methors of `DelayedIconDelegate` are called even before queued method calls?

Reply:
> Did you test this w/o...

I never make a PR without testing.